README.md

```diff
@@ -217,9 +218,20 @@ const name = 'Eames Chair'
 const popup = false
 const progressBarPosition = 'middle'
 const poster = 'https://real2u-public-assets.s3.amazonaws.com/images/cadeira.png'
+const buttonVariant = document.getElementById('button-variant1')
+const variants = [{ element: buttonVariant, sku: 'RE000002' }]
 
-R2U.viewer.create({ element, sku, name, popup, progressBarPosition, poster })
+R2U.viewer.create({ element, sku, name, popup, progressBarPosition, poster, variants})
 ```
+variants[]
+
+| parameter             | description                                                              | default                      |
+| --------------------- | ------------------------------------------------------------------------ | ---------------------------- |
+| `element`             | HTML element that will receive the 3D viewer                             | `''`                         |
```

Comment:
isso tbm é estranho, porque o `element` já tá no `R2U.viewer.create`

Reply:
Acho que o que precisa ser pensado é numa estrutura que faça sentido então. A ideia é que o viewer tenha associado a ele, outros elementos que possam alterar o SKU sem criar outro viewer.
O jeito que pensamos foi um array de objetos, cada objeto carrega um elemento HTML pra disparar uma troca, o SKU novo e o tipo de evento que dispara a troca. Se isso não fizer sentido na documentação, acho que vale pensar numa forma mais agradável de entender esse conceito.
@aviggiano , acha que faz mais sentido voltar a ter uma função que realiza a troca recebendo apenas o SKU e deixar os eventos do lado do cliente?
cc @MayaraSs - Pensar nessa legibilidade, quando eu pensei na estrutura não pensei muito nisso, acho que vc pode trabalhar melhor nisso com esse feedback do Antonio.

Reply:
> *A ideia é que o viewer tenha associado a ele, outros elementos que possam alterar o SKU sem criar outro viewer.*

Essa informação acima é bem killer no design da documentação!

Acho que tem que ser algo nessa linha. Por exemplo

```typescript
const handle = R2U.viewer.create({...})
handle.sku = newSku

// poderia ser usado da seguinte maneira
const triggers = document.getElementsByClassName('list-sku')
triggers.addEventListener('click', e => handle.sku = e.name)
```
É só uma ideia, mas ela é parecida com a abordagem de "handles" do puppeteer: https://pub.dev/documentation/puppeteer/latest/puppeteer/JsHandle-class.html

Um "handle" é um objeto que uma API expõe para te dar acesso de alterar o comportamento interno daquela API.
Algo nessa linha: https://en.wikipedia.org/wiki/Chain-of-responsibility_pattern

Reply:
Ideia 2:

```typescript
const handle = R2U.viewer.create({...})

// get triggers to change variant
const triggers = document.getElementsByClassName('list-sku')
triggers.addEventListener('click', e => handle.setSku(e.target.value))
```

Reply:
Ideia 3:

```
// usar hooks
```

Reply:
```typescript
const hook = setSkuState(...)

const triggers = document.getElementsByClassName('list-sku')
triggers.addEventListener('click', e => hook(e.target.value))

interface ViewerHandle {
  setSkuState: (sku: string) => Promise<void>
}

// no codigo
const [sku, setSku] = useState('')
const setSkuStat = (sku) => { 
// chama o setSku do state
setSku(...)
}

useEffect(() => {.... //chama a funcao }, [sku]) 
```

Reply:
Achei interessante @MayaraSs 

Meu único "porém" entre a opção 3 (react hooks) e a opção 2 (função setter exposta pelo SDK) é que a 3 necessita que o cliente integrador esteja usando react. Seria possível nós mesmos "criarmos" uma lógica de hooks que imita o react hoots, mas acho meio overkill. Por esse motivo, eu prefiro a opção 2.

Reply:
entendii, verdade